packages/snap/src/index.ts

```diff
@@ -1,104 +1,115 @@
+  'getBlock',
+  'getBalance',
+  'getChainHead',
+  'signPayloadJSON',
+  'signPayloadRaw',
+  'generateTransactionPayload',
+  'send'
 ];
-// eslint-disable-next-line
-export const onRpcRequest: OnRpcRequestHandler = (async ({ request }) => {
+
+export const onRpcRequest: OnRpcRequestHandler = async ({ request }) => {
   const state = await snap.request({
     method: 'snap_manageState',
-    params: { operation: 'get' },
+    params: { operation: 'get' }
   });
 
   if (!state) {
     // initialize state if empty and set default config
     await snap.request({
       method: 'snap_manageState',
-      params: { newState: EmptyMetamaskState(), operation: 'update' },
+      params: { newState: EmptyMetamaskState(), operation: 'update' }
     });
   }
   // fetch api promise
   let api: ApiPromise = null;
   if (apiDependentMethods.includes(request.method)) {
-    api = await getApi(snap);
+    api = await getApi();
   }
 
   switch (request.method) {
-    case "signPayloadJSON":
+    case 'signPayloadJSON':
       assert(request.params, validSignPayloadJSONSchema);
-      return await signPayloadJSON(snap, api, request.params.payload);
-    case "signPayloadRaw":
+      return await signPayloadJSON(api, request.params.payload);
+    case 'signPayloadRaw':
       assert(request.params, validSignPayloadRawSchema);
-      return await signPayloadRaw(snap, api, request.params.payload);
+      return await signPayloadRaw(api, request.params.payload);
     case 'getPublicKey':
-      return await getPublicKey(snap);
+      return await getPublicKey();
     case 'getAddress':
-      return await getAddress(snap);
+      return await getAddress();
     case 'exportSeed':
-      return await exportSeed(snap);
+      return await exportSeed();
     case 'getAllTransactions':
-      return await getTransactions(snap);
+      return await getTransactions();
     case 'getBlock':
       assert(request.params, validGetBlockSchema);
       return await getBlock(request.params.blockTag, api);
     case 'getBalance': {
-      return await getBalance(snap, api);
+      return await getBalance(api);
     }
     case 'configure': {
-      const state = await snap.request({
+      const state = (await snap.request({
         method: 'snap_manageState',
-        params: { operation: 'get' },
-      }) as MetamaskState;
-      const isInitialConfiguration = state.polkadot.config === null;
+        params: { operation: 'get' }
+      })) as MetamaskState;
+
+      const isInitialConfiguration = state.config === null;
       // reset api and remove asset only if already configured
       if (!isInitialConfiguration) {
         resetApi();
       }
       // set new configuration
-      assert(request.params, validConfigureSchema);
+      assert(
+        request.params,
+        validConfigureSchema,
+        'Invalid configuration schema - Network name should be provided'
```

Comment:
could we allow developer to pass network config?

Reply:
You can pass the whole config. The only requirement is to provide at least network name